src/QsCompiler/Transformations/CallGraphWalker.cs

```diff
@@ -123,39 +150,51 @@ public override ExpressionKind OnIdentifier(Identifier sym, QsNullable<Immutable
                 {
                     // Type arguments need to be resolved for the whole expression to be accurate
                     // ToDo: this needs adaption if we want to support type specializations
-                    var typeArgs = tArgs;
+                    var typeArgs = QsNullable<ImmutableArray<ResolvedType>>.Null;
```

Comment:
Here would be an implementation for populating the type argument handles that are currently unused: https://github.com/microsoft/qsharp-compiler/pull/456
The idea would be that the cycle check would be run for each library and the source code separately, and then the full call graph containing the full type information can be build and stored with the compilation (just watch out to keep what is stored with the compilation immutable).

Reply:
I don't understand what this comment is addressing or what the linked implementation does. What is the issue you are trying to address here?

Reply:
The todo's in the code.

Reply:
I'm going to wait to make changes on this topic until it can be worked into the design doc for the call graph. I will create a separate task for this and its implementation will be a separate PR. For now, this PR needs to be merged in so that work can continue on the feature.